bundled-guava/src/main/java/org/apache/iceberg/GuavaClasses.java

```diff
@@ -98,5 +99,6 @@ public class GuavaClasses {
     CountingOutputStream.class.getName();
     Suppliers.class.getName();
     Stopwatch.class.getName();
+    GcFinalization.class.getName();
```

Comment:
@rdblue any thoughts on adding a test class to `GuavaClasses` vs relaxing checkstyle so that classes from `com.google.common.testing` are allowed in tests?

Reply:
I'd probably allow the direct classes in tests.

Reply:
@rizaon for this you'd want to add something to our [checktyle rules](https://github.com/apache/iceberg/tree/master/.baseline/checkstyle) so that classes from `com.google.common.testing` are allowed to be used in tests directly rather than having to bundle them via `GuavaClasses`

Reply:
Done in 3063b1a

Reply:
Using the suppression like this would allow people to e.g. use `com.google.common.base.Preconditions` directly in tests right? We still want to make sure that classes from `com.google.common` are used via `org.apache.iceberg.relocated.*` in tests as well. 
You might need to play a bit with the correct checkstyle configuration so that
* all java code can't import `com.google.common` directly
* test java code can import `com.google.common.testing` only